### PR TITLE
Generator NG

### PR DESCRIPTION
**Pull request by jirikuncar**
Port to Jinja2

### CODE REVIEW

File: lib/datadog_api_client/v1/models/creator.rb [hunk 1]
```diff
@@ -57,7 +57,6 @@ def self.openapi_types
     # List of attributes with nullable: true
     def self.openapi_nullable
       Set.new([
-        :'name'
```
Comment (therve) [inline lib/datadog_api_client/v1/models/creator.rb]: Looks like an issue?

File: lib/datadog_api_client/v1/api/hosts_api.rb [hunk 2]
```diff
@@ -135,6 +144,15 @@ def list_hosts_with_http_info(opts = {})
...
       end
+
+
+
+
+
+
+
+
+
```
Comment (therve) [inline lib/datadog_api_client/v1/api/hosts_api.rb]: Would be nice to clean that out.

File: .generator/pyproject.toml [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+python = "^3.9"
+click = "8.0.1"
+PyYAML = "6.0"
+jsonref = "0.2"
+jinja2 = "3.0.3"
+
+[tool.poetry.dev-dependencies]
+
+[build-system]
+requires = ["poetry-core>=1.0.0"]
```
Comment (therve) [inline .generator/pyproject.toml:18]: Do we have to use poetry for this? just wondering as we've done without until now.

File: .generator/src/generator/openapi.py [hunk 4]
```diff
@@ -0,0 +1,432 @@
...
+def response(operation, status_code=None):
+    for response in operation["responses"]:
+        if status_code is None or response == str(status_code):
+            return list(operation["responses"][response]["content"].values())[0][
+                "schema"
+            ]
+    return None
+
+
+def lookup(data, dotted_path):
```
Comment (therve) [inline .generator/src/generator/openapi.py:337]: I don't think we need this.

File: .generator/src/generator/openapi.py [hunk 5]
```diff
@@ -0,0 +1,432 @@
...
+        raise TypeError(f"Unknown type: {spec['type']}")
+
+
+class Schema:
+    def __init__(self, spec, value=None, keys=None):
+        self.spec = spec
+        self.value = value if value is not None else generate_value
+        self.keys = keys or tuple()
+
+    def __getattr__(self, key):
```
Comment (therve) [inline .generator/src/generator/openapi.py:394]: Can we get rid of this method?

**Review by therve — COMMENTED** [review]
Looks really nice! Only spotting some small issues like nullable, and constraints check missing. It'd be nice to remove docs and spec in a different PR to reduce the diff (Chrome is dying as I speak).

We need to think about a solution for docs like rdoc or yard.

**Review by therve — APPROVED** [review]

File: .generator/src/generator/templates/api.j2 [hunk 6]
```diff
@@ -0,0 +1,223 @@
...
+  {%- endif %}
+{%- endfor %}
+      # resource path
+      local_var_path = '{{ path }}'{% for parameter in operation.parameters if parameter.in == "path" %}.sub('{{ '{' }}{{ parameter.name }}{{ '}'}}', CGI.escape({{ parameter.name }}.to_s).gsub('%2F', '/')){% endfor %}
+
+      # query parameters
+      query_params = opts[:query_params] || {}
+      {%- for name, parameter in operation|parameters if parameter.in == "query" and parameter.required %}
+      {%- set collection = parameter|parameter_schema|collection_format %}
+      query_params[:'{{ name }}'] = {% if collection %}@api_client.build_collection_param({{ name|attribute_name }}, :{{ collection }}){% else %}{{ name|attribute_name }}{% endif %}
```
Comment (therve) [inline .generator/src/generator/templates/api.j2:146]: Something is wrong around that logic, not sure what.

Comment (jirikuncar) [reply .generator/src/generator/templates/api.j2:146]: solved

**Pull request merged**